fuel-p2p/src/orchestrator.rs

```diff
@@ -0,0 +1,98 @@
+use fuel_core_interfaces::model::Vote;
+use fuel_tx::Transaction;
+use libp2p::identity::Keypair;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+use crate::{
+    behavior::FuelBehaviourEvent,
+    config::P2PConfig,
+    gossipsub::messages::GossipsubMessage,
+    service::{FuelP2PEvent, FuelP2PService},
+};
+
+pub struct NetworkOrchestrator {
+    p2p_service: FuelP2PService,
+
+    // receivers
+    rx_block_producer: Receiver<()>,
+    rx_bft: Receiver<Vote>,
+
+    // senders
+    tx_synchronizer: Sender<()>,
+    tx_transaction_pool: Sender<Transaction>,
+    tx_bft: Sender<()>,
```

Comment:
Also don't understand this `()`. Is it WiP?

Reply:
@vlopes11 yes, it's a work in progress :)